### PR TITLE
Fastspkt

### PR DESCRIPTION
**Pull request by jjcarre**
- Added a getSpktInclude function to analysis.py, it returns a dataframe containing spkt and spkid.
- Refactored plotRaster to use pandas data frames to speed it up.

### CODE REVIEW

**Comment by salvadord** [comment]
@jjcarre is this tested and ready to merge? if so can you try to resolve the conflicts in analysis.py (another PR was also making changes in analysis) ?

**Comment by jjcarre** [comment]
Yes, I've tested it on the examples and resolved the conflicts.

File: netpyne/analysis.py [hunk 1]
```diff
@@ -596,91 +615,87 @@ def plotRaster (include = ['allCells'], timeRange = None, maxSpikes = 1e8, order
         - syncLines (True|False): calculate synchorny measure and plot vertical lines for each spike to evidence synchrony (default: False)
         - lw (integer): Line width for each spike (default: 2)
         - marker (char): Marker for each spike (default: '|')
-        - popColors (dict): Dictionary with color (value) used for each population (key) (default: None)
+        - popColors (odict): Dictionary with color (value) used for each population (key) (default: None)
```
Comment (salvadord) [inline netpyne/analysis.py]: @jjcarre why odict?

File: netpyne/analysis.py [hunk 2]
```diff
@@ -596,91 +615,87 @@ def plotRaster (include = ['allCells'], timeRange = None, maxSpikes = 1e8, order
...
-        elif isinstance(orderBy, list) and len(orderBy) == 2:
-            yorders = [[popLabels.index(cell['tags'][orderElem]) if orderElem=='pop' else cell['tags'][orderElem] 
-                                    for cell in cells] for orderElem in orderBy] 
-            #sortedGids = {gid:i for i,  in enumerate(sorted(zip(yorders[0], yorders[1], cellGids)))}
-            sortedGids = [gid for (y0, y1, gid) in sorted(zip(yorders[0], yorders[1], cellGids))]
-
-        #sortedGids = {gid:i for i, (y, gid) in enumerate(sorted(zip(yorder, cellGids)))}
-        spkinds = [sortedGids.index(gid)  for gid in spkgids]
+    if len(df) > 0:
+        ylabelText = 'Cells (ordered by %s)'%(orderBy)
```
[the remark this answers is not in the document]
Comment (jjcarre) [reply netpyne/analysis.py]: Yes the DataFrame.sort_values method can accept a list as an argument

File: netpyne/analysis.py (hunk 1 above)
Comment (jjcarre) [reply netpyne/analysis.py]: I didn't make that change, but there is nothing in the old or new plotRaster that enforces an odict

**Pull request merged**